src/array.jl

```diff
@@ -236,26 +224,27 @@ CategoricalMatrix(A::CategoricalArray{T, 2, R};
 
 # From AbstractArray
 convert(::Type{CategoricalArray{T, N}}, A::AbstractArray{S, N}) where {S, T, N} =
-    convert(CategoricalArray{T, N, DefaultRefType}, A)
+    convert(CategoricalArray{T, N, reftype(T)}, A)
 convert(::Type{CategoricalArray{T}}, A::AbstractArray{S, N}) where {S, T, N} =
     convert(CategoricalArray{T, N}, A)
 convert(::Type{CategoricalArray}, A::AbstractArray{T, N}) where {T, N} =
     convert(CategoricalArray{T, N}, A)
 
+# FIXME T=String would use wrong categorical value, also it's not extensible, where it's used?
```

Comment:
Are these methods still needed since the constructor above already ensured we don't wrap `CategoricalValue` inside `CategoricalValue`? Removing them would fix the problem. :-)